src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -24,6 +24,9 @@ internal class TupleFieldSymbol : WrappedFieldSymbol
         public TupleFieldSymbol(TupleTypeSymbol container, FieldSymbol underlyingField, int tupleFieldId)
             : base(underlyingField)
         {
+            Debug.Assert(container.UnderlyingNamedType.Equals(underlyingField.ContainingType, TypeCompareKind.IgnoreDynamicAndTupleNames) || this is TupleVirtualElementFieldSymbol,
```

Comment:
I'd prefer we avoid depending on the identity of particular internal implementation classes. Can `this is TupleVirtualElementFieldSymbol` be replaced by `this.IsVirtualTupleField`?